src/sysfiles.c

```diff
@@ -3540,41 +3527,6 @@ Char *SyTmpname ( void )
   return name;
 }
 
-#else
-Char * SyTmpname ( void )
-{
-    static Char   * base = 0;
-    static Char     name[1024];
-    static UInt     count = 0;
-    Char          * s;
-    UInt            c;
-
-    SyClearErrorNo();
-    count++;
-    if ( base == 0 ) {
-        base = tmpnam( (char*)0 );
-    }
-    if ( base == 0 ) {
-        SySetErrorNo();
-        return 0;
-    }
-    if (count == 0) { /* one time in 2^32 */
-        strcat( base, "x" ); /* FIXME: BUG, we must not modify the pointer returned by tmpnam */
```

Comment:
Note the FIXME here; getting rid of this was the original motivation for this PR.